src/SurfaceFluxes.jl

```diff
@@ -416,31 +415,31 @@ function obukhov_length(
         L_MO = sol.root
         if !sol.converged
             if error_on_non_convergence()
-                KA.@print("maxiter reached in SurfaceFluxes.jl:\n")
-                KA.@print(" T_in = ", TD.air_temperature(thermo_params, ts_in(sc)))
-                KA.@print(", T_sfc = ", TD.air_temperature(thermo_params, ts_sfc(sc)))
-                KA.@print(", q_in = ", TD.total_specific_humidity(thermo_params, ts_in(sc)))
-                KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
-                KA.@print(", u_in = ", u_in(sc))
-                KA.@print(", u_sfc = ", u_sfc(sc))
-                KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
-                KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
-                KA.@print(", Δz = ", Δz(sc))
-                KA.@print(", ΔDSEᵥ = ", ΔDSEᵥ)
-                KA.@print("\n")
-                KA.@print("ts_in(sc) = ", ts_in(sc))
-                KA.@print("\n")
-                KA.@print("ts_sfc(sc) = ", ts_sfc(sc))
-                KA.@print("\n")
+                @printf("maxiter reached in SurfaceFluxes.jl:\n")
```

Comment:
The KernelAbstractions `@print` is GPU compatible. And my guess is that`@printf` is not

Reply:
Right - for now this PR switches from KA -> CUDA + ClimaComms, the idea was to re-introduce the GPU print in the next PR. Would [CUDA.@cuprintf](https://docs.juliahub.com/CUDA/oWw5k/1.2.1/api/kernel/#CUDA.@cuprintf) this be a reasonable alternative ?

Reply:
Maybe, but the KernelAbstractions version adapts between CPU and GPU